cli/tests/test_next_prev_commands.rs

```diff
@@ -567,6 +567,127 @@ fn test_next_editing() {
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_next_conflict() {
+    // There is a conflict in the third commit, so after next it should be the new
+    // parent.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    std::fs::write(&file_path, "Other text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // Create a conflict in the third commit.
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(second)"]);
+    std::fs::write(&file_path, "some text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "description(third)"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
+    test_env.jj_cmd_ok(&repo_path, &["edit", "@---"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["next", "--conflict"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: kkmpptxz 9de45eb7 (conflict) third
+    Parent commit      : rlvkpnrz 374aea34 second
+    Added 0 files, modified 1 files, removed 0 files
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+    // --edit is implied when already editing a non-head commit
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["next"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Working copy now at: royxmykx 311dd604 (conflict) (empty) fourth
+    Parent commit      : kkmpptxz 9de45eb7 (conflict) third
+    There are unresolved conflicts at these paths:
+    content.txt    2-sided conflict
+    "###);
+}
+
+#[test]
+fn test_next_conflict_editing() {
+    // There is a conflict in the third commit, so after next it should be our
+    // working copy.
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let file_path = repo_path.join("content.txt");
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
+    std::fs::write(&file_path, "Other text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // Create a conflict in the third commit.
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    std::fs::write(&file_path, "Sir, we've got a problem").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["edit", "description(second)"]);
+    std::fs::write(&file_path, "some text").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "@+ ~ @"]);
+    test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "fourth"]);
```

Comment:
IIUC, this fourth commit has no effect on the test case, so delete it?

Reply:
removed.